src/test/java/com/aws/greengrass/lifecyclemanager/LogManagerHelperTest.java

```diff
@@ -123,9 +131,122 @@ void GIVEN_mock_service_WHEN_getComponentLogger_THEN_logs_to_correct_log_file()
         assertEquals(0, ggLogFile.length());
     }
 
+    @Test
+    void GIVEN_mock_service_logger_WHEN_reconfigure_THEN_change_applied_correctly() throws IOException {
+        Path tempRootDir2 = tempRootDir.resolve("2");
+        Path tempRootDir3 = tempRootDir.resolve("3");
+        String mockServiceName = "MockService001";
+        when(mockGreengrassService.getServiceName()).thenReturn(mockServiceName);
+        LogConfig.getInstance().setStore(LogStore.FILE);
+        Logger componentLogger = LogManagerHelper.getComponentLogger(mockGreengrassService);
+
+        // change log file size
+        LogConfigUpdate newConfig = LogConfigUpdate.builder().fileSizeKB(1L).build();
+        LogManager.reconfigureAllLoggers(newConfig);
+
+        // should apply change to all loggers
+        LogConfig testLogConfig = LogManager.getLogConfigurations().get(mockServiceName);
+        assertEquals(1, testLogConfig.getFileSizeKB());
+        assertEquals(1, LogManager.getRootLogConfiguration().getFileSizeKB());
+        assertEquals(1, LogManager.getTelemetryConfig().getFileSizeKB());
+
+        // log less than 1k bytes, should not rotate
+        // 1 message of size 50 is about 128 bytes in TEXT format
+        logRandomMessages(componentLogger, 50, 1);
+        assertEquals(1, getLogFileCount(testLogConfig, mockServiceName));
+
+        // log more than 2k bytes. Should rotate this time
+        logRandomMessages(componentLogger, 50, 20);
+        assertTrue(getLogFileCount(testLogConfig, mockServiceName) > 1);
```

Comment:
The rolling limit is 1KB... Is logging 2KB too much for the test? Ideally 1025 byte :p

Reply:
fixed